Update to 21.11.0

leej3 opened the pull request:
+ regenerate patch due to failed application (I'm not sure if this was the best approach... advice welcome).
+ add boost-cpp for splash (this could instead to be configured to remove dependency).
+ ninja is now default build system for the project (alternatively the cmake command could be modified to generate a make build system).

leej3 committed:
update version

regenerate patch due to failed application.
add boost-cpp for splash.
ninja is now default build system for the project.

katietz commented on recipe/meta.yaml:
```diff
@@ -40,6 +40,7 @@ requirements:
     - {{ compiler('cxx') }}
     - pkg-config  # [not win]
     - cmake
+    - ninja
```
is ninja used also on Unix here, or just on Windows?

leej3 replied:
yes. I didn't look hard for where ninja was specified as the build system used by cmake. One can set it in the cmake code or you as part of the cmake command itself: `cmake -GNinja ...` or something like `cmake -G'Unix Makefiles'`. In general I think ninja is a little faster and provides a tremendous speed increase when re-running a build with small changes to the source. I suspect the latter is somewhat irrelevant here.

In this case I just included ninja as a dependency so that I didn't have to figure out where they had changed that in their cmake files.

katietz replied:
ok, we just should keep in mind. thanks

katietz commented on recipe/meta.yaml:
```diff
@@ -40,6 +40,7 @@ requirements:
     - {{ compiler('cxx') }}
     - pkg-config  # [not win]
```
you know that pkg-config is available on Windows too?

leej3 replied:
I didn't. I didn't test this for anything but linux though. The "mac-volatile..." patch does not successfully apply so those builds fail. And on s390x the build fails because boost is not available as a dependency (boost would now be a dependency unless a little more work is done to configure cmake to build without the boost dependent splash functionality).

EDIT: removing the mac patch does result in subsequent errors... so more work required if we want a release on that architecture

katietz replied:
the mac-volatile patch is a giant hack to work-a-round a clang issue on casting of volatiles.  I hope we won't need it for newer version anymore, so it would be good if it got no longer required

katietz reviewed (approved):
With those two questions answered, patch looks good to me

leej3 committed:
tidy build.sh

various flags are set automatically during compiler activation in
CMAKE_ARGS, which is now passed to cmake

Set Ninja as the build generator

leej3 committed:
remove mac_volatile.patch

leej3 committed:
remove tests dependent on conda-build

leej3 committed:
define CMAKE_ARGS on ppc64le and skip s390x
